test/CMakeLists.txt

```diff
@@ -43,36 +47,28 @@ execute_process(COMMAND ${CMAKE_COMMAND} --build .
 # the following targets: gtest, gtest_main, gmock
 # and gmock_main
 add_subdirectory(${CMAKE_BINARY_DIR}/googletest-src
-${CMAKE_BINARY_DIR}/googletest-build)
-
-# The gtest/gmock targets carry header search path
-# dependencies automatically when using CMake 2.8.11 or
-# later. Otherwise we have to add them here ourselves.
-include_directories("${gtest_SOURCE_DIR}/include"
-    "${gmock_SOURCE_DIR}/include")
+${CMAKE_BINARY_DIR}/googletest-build EXCLUDE_FROM_ALL)
 
-add_definitions(-DENABLE_TESTS -DPLATFORM_AGNOSTIC_BUILD)
 enable_testing()
 
-# GTest and GMock do not play nicely with these flags, so disable them
-target_compile_options(gtest PRIVATE -Wno-sign-promo -Wno-pedantic)
-target_compile_options(gmock PRIVATE -Wno-conversion -Wno-pedantic)
+# Link against gtest library
+target_link_libraries (TestCommon INTERFACE gtest gmock_main)
```

Comment:
How come we don't need to ignore this warnings anymore? Or is this done somewhere that I've missed?

Reply:
We no longer modify `CMAKE_CXX_FLAGS`, so the gtest/gmock targets are no longer initialized with their compilation flags property full of our warnings.

Reply:
Oh right, I forgot those were the global flags :sweat_smile: